frontend/src/components/PetitionCard.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Dispatch,
+  FormEventHandler,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { FaSpinner } from "react-icons/fa";
+import { PetitionDetail, Question } from "../models/Petition";
+import { submitPetitionSignature } from "../utils/backend";
+
+interface Props {
+  className?: string;
+  petition: PetitionDetail;
+  link?: ReactNode;
+}
+
+enum State {
+  Initial,
+  Submitting,
+  Error,
+  Existed,
+  Success,
+}
+
+const SESSION_KEY = "petitionData";
+
+function makeBox(
```

Comment:
Kan je hier een normaal functional component van maken, zoals gebruikelijk volgens de React-conventies?
https://www.robinwieruch.de/react-function-component/#react-function-component-example

Kijk ook naar component composition. Dat is de gangbare manier hier je specifiek gedrag introduceert in generieke componenten (een specifieke PetitionBox-state, bijvoorbeeld).